subt/artifacts.py

```diff
@@ -308,6 +340,7 @@ def update(self):  # hack, this method should be called run instead!
                 s = '%s %.2f %.2f %.2f\n' % (artf_type, ix/1000.0, iy/1000.0, iz/1000.0)
                 f.write(s)
                 self.publish('artf_cmd', bytes('artf ' + s, encoding='ascii'))
+                self.sleep(3.0)  # make sure it is sent and processed!
```

Comment:
Why does the `sleep` call help to make sure it is sent and processed?

Reply:
Well, it is simulation BUT it looked like they have similar "throttling" (?) filter like DARPA in System Track, i.e. that you cannot report many artifacts in short period of time. The sleep helped but does not guarantee delivery. (that part would be necessary on C++ side)